### ISSUE TITLE
Fails to jump to load jupyter symlink when notebook isn't in root folder

### ISSUE THREAD

**Issue by thoralf-gutierrez** [krassowski/jupyterlab-go-to-definition#16]
Hello!

I'm running into the following issue. When I am working on a notebook that is not in my root folder, e.g. if I'm working on `subfolder/hello.ipynb`.

When I am inside `hello.ipynb` and I try to go to the definition of a function that is outside of my notebook, I see the relevant file appear inside `.jupyter_symlinks`, and my file tree looks like

```
* file_at_root.txt
* subfolder/hello.ipynb
* subfolder/.jupyter_symlinks/path/to/file.py
```

When I try to go to the external definition, it fails to jump, because this api call returns 404
* jupyterhub.local/user/my_username/api/contents/.jupyter_symlinks/path/to/file.py

But the following api call would have worked
* jupyterhub.local/user/my_username/api/contents/subfolder/.jupyter_symlinks/path/to/file.py

So it looks like either the `.jupyter_symlinks` folder should always be at the root of my jupyter folder, or the path in the URL should include the base path of the notebook itself.

Are you seeing the same thing?

**Comment by krassowski** [comment]
Yes, `.jupyter_symlinks` should always be at the root of your Jupyter (i.e. where you start JupyterLab from). Apologies if this was not clearly explained in the docs. Is there a reason why you would prefer not to have it in the Jupyter project root?

**Comment by thoralf-gutierrez** [comment]
I don't think I did anything special, the `.jupyter_symlinks` just appeared in the subfolder instead of the root folder. Maybe that's the bug then?

**Comment by krassowski** [comment]
It might be. However, this would be very interesting as normally, the user needs to [create the `.jupyter_symlinks` folder manually](https://github.com/krassowski/jupyterlab-go-to-definition#symlink-workaround-jump-to-any-file-outside-of-the-project-root). I do not expect it to be created by the extension - it would be useless as it would not know where to symlink to...

Would you mind providing a concrete reproducible example (i.e. where is the file with the definition)?

**Comment by thoralf-gutierrez** [comment]
For me the `.jupyter_symlinks` appears on its own, I've never had to create one.

Reproducible example from home folder that works
* from the terminal, in jupyter home folder, do`rm -rf .jupyter_symlinks/`
* create a new notebook `Untitled.ipynb`
* add `import json`
* put cursor on `json` and CTRL + ALT + B and successfully jump to `.jupyter_symlinks/usr/lib/python3.5/json/__init__.py`
* notice that `.jupyter_symlinks/usr/lib/python3.5/json/__init__.py` was created automatically

Reproducible example from subfolder that does not work
* from the terminal, in jupyter home folder, do`rm -rf .jupyter_symlinks/` <-- **clean up the previous experiment**
* then, `mkdir example`
* create a new notebook `example/Untitled.ipynb`
* add `import json`
* put cursor on `json` and CTRL + ALT + B notice the failure in your console log, 404 for https://jupyterhub.host/user/myuser/api/contents/.jupyter_symlinks/usr/lib/python3.5/json/__init__.py?content=0&1575587453311
* notice though that a `example/.jupyter_symlinks/usr/lib/python3.5/json/__init__.py` was created automatically

**Comment by caizefeng** [comment]
Exactly the same problem there.
